WaitForSnapshotStepTests#testHashcodeAndEquals fails

Failed on my backport PR: https://gradle-enterprise.elastic.co/s/w35l6ouk6uzsw and reproducible on 7.x.

```
./gradlew ':x-pack:plugin:core:test' --tests "org.elasticsearch.xpack.core.ilm.WaitForSnapshotStepTests.testHashcodeAndEquals" -Dtests.seed=F61F844292A93219 -Dtests.security.manager=true -Dtests.locale=fr-BE -Dtests.timezone=Australia/South -Dcompiler.java=13 -Druntime.java=8
```

```
org.elasticsearch.xpack.core.ilm.WaitForSnapshotStepTests > testHashcodeAndEquals FAILED
    java.lang.AssertionError: WaitForSnapshotStep mutation should not be equal to original
    Expected: not <{"phase":"SmZLaEYALs","action":"rvBecejUXU","name":"JvfEvWSTLz"} => {"phase":"OvCHIYVrPd","action":"uWOZHJNBGg","name":"EUaKQDgdvB"}>
         but: was <{"phase":"SmZLaEYALs","action":"rvBecejUXU","name":"JvfEvWSTLz"} => {"phase":"OvCHIYVrPd","action":"uWOZHJNBGg","name":"EUaKQDgdvB"}>
```

--- a/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/WaitForSnapshotStepTests.java
+++ b/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/WaitForSnapshotStepTests.java
@@ -41,7 +41,7 @@ public class WaitForSnapshotStepTests extends AbstractStepTestCase<WaitForSnapsh
                 nextKey = new Step.StepKey(key.getPhase(), key.getAction(), key.getName() + randomAlphaOfLength(5));
                 break;
             case 2:
-                policy = randomAlphaOfLengthBetween(1, 10);
+                policy = randomValueOtherThan(policy, () -> randomAlphaOfLengthBetween(1, 10));
                 break;
             default:
                 throw new AssertionError("Illegal randomisation branch");

